src/libtorch.cc

```diff
@@ -552,11 +582,48 @@ ModelInstanceState::Create(
 
 ModelInstanceState::ModelInstanceState(
     ModelState* model_state, TRITONBACKEND_ModelInstance* triton_model_instance)
+#ifdef TRITON_ENABLE_GPU
+    : BackendModelInstance(model_state, triton_model_instance),
+      model_state_(model_state), device_(torch::kCPU), is_dict_input_(false),
+      pytorch_stream_(at::cuda::getDefaultCUDAStream())
+#else
     : BackendModelInstance(model_state, triton_model_instance),
       model_state_(model_state), device_(torch::kCPU), is_dict_input_(false)
+#endif
 {
   if (Kind() == TRITONSERVER_INSTANCEGROUPKIND_GPU) {
     device_ = torch::Device(torch::kCUDA, DeviceId());
+#ifdef TRITON_ENABLE_GPU
+    pytorch_stream_ =
+        at::cuda::getStreamFromPool(false /* is_high_priority */, DeviceId());
```

Comment:
From the documentation it looks like this call only returns cuda stream from a pre-allocated pool. Where are the cuda streams in the pool being created?
https://pytorch.org/cppdocs/api/function_namespacec10_1_1cuda_1a42eb10631a7944924acd387aeae13245.html

I think it will make more sense to use high_priority cuda stream for inference if the stream is already been created in the pool. Have you run your experiments with is_high_priority=true?
I wonder if we can expose that as an option in model config.

Reply:
The high priority option looks like it gets a stream from another pool, is this high priority pool a part of the same pre-allocation? Are these high priority streams local to the model instance or global to the device?

Reply:
Switched to model instance CUDA instances instead of using CUDA . I have not run any experiment using the is_high_priority=true.

> Where are the cuda streams in the pool being created?

It is created in PyTorch: https://github.com/pytorch/pytorch/blob/f9ef363982136f45dfb2bd4205c545cb17e59afd/c10/cuda/CUDAStream.cpp#L23-L50
Looks like there are 32 streams per GPU per pool.

> Have you run your experiments with is_high_priority=true?

I have not run the experiments with high priority=true. It looks like TRT also uses CUDAStreams with priorities? I was wondering whether you have seen any speedup gains from using streams in TRT. https://github.com/triton-inference-server/tensorrt_backend/blob/main/src/tensorrt.cc#L3064-L3077

I can file a follow up ticket to investigate this and expose this option in backend utilities or model configuration.